### ISSUE TITLE
About dictionaries, can two items associate to the same value?

### ISSUE THREAD

**Issue by GhbSmwc** [godotengine/godot-docs#4487]
Reading [this article](https://docs.godotengine.org/en/stable/classes/class_dictionary.html) it stated “Dictionaries are composed of pairs of keys (which must be unique) and values.”

So I was wondering when it meant “unique” is that each one item in the pair (A) must be associated with only ONE of the other item in the pair (B).

However, can two different items be associated to the same, while it cannot have one item associated to multiple items? (like being akin to [function in mathematics with ordered pairs](https://en.wikipedia.org/wiki/Function_(mathematics)#Definition))

Something like this:
```
var my_dir = {} # Creates an empty dictionary.
var points_dir = {"White": 50, "Yellow": 50, "Orange": 50}
```
But you cannot do this:
```
var my_dir = {} # Creates an empty dictionary.
var points_dir = {"White": 50, "White": 75, "Orange": 50}
```

If yes, someone should update that document to say this instead:

“Dictionary type. Associative container which contains values referenced by unique keys. Dictionaries are composed of pairs of things; a key (which must be unique) and a value. Dictionaries will preserve the insertion order when adding elements, even though this may not be reflected when printing the dictionary. In other programming languages, this data structure is sometimes referred to as an hash map or associative array.”

**Comment by pycbouh** [comment]
It's a bit more complicated than that, because "same" value can be understand differently depending on how familiar you are with programming concepts and the issue you are trying to solve. In your examples, technically, all values are unique, but equal. Those are different values, but their type is the same and their quality can be compared and deduced to be equal. However, it only works like that when you talk about variables passed by value. If you were to store references to objects instead different keys could indeed refer to the same exact value, not just equal but unique values.

So your suggested change can also be misleading.

In short, this is a pretty generic programming pattern, an associative collection where each record has a unique key, i.e. that record's "address" in this collection, and some value associated with it, which can be anything.

**Comment by Omicron666** [comment]
> Reading [this article](https://docs.godotengine.org/en/stable/classes/class_dictionary.html) it stated “Dictionaries are composed of pairs of keys (which must be unique) and values.”

This is preceeded with "Associative container which contains values referenced by unique keys."
It would say "unique values" if that was the case.

Concerning same page. I have doubt about the assertion "Dictionaries will preserve the insertion order when adding elements, even though this may not be reflected when printing the dictionary"
Isn't it a bit confusing?
